tests/infra/github.py

```diff
@@ -291,3 +323,105 @@ def install_next_lts_for_branch(self, branch):
 
         LOG.info(f"Next release tag: {next_tag}")
         return self.install_release(next_tag)
+
+
+if __name__ == "__main__":
+    # Run this to test
+    class MockGitEnv:
+        def __init__(self, tags, release_branches):
+            self.tags = tags
+            self.release_branches = release_branches
+
+        def has_release_for_tag_name(self, tag_name):
+            return True
+
+    def make_test_vector(tags, release_branches, local_branch, expected):
+        return {
+            "tags": tags,
+            "release_branches": release_branches,
+            "local_branch": local_branch,
+            "expected": expected,
+        }
+
+    test_scenarios = [
+        # Development on main after release 1.x
+        make_test_vector(
+            ["ccf-1.0.0", "ccf-1.0.1"],
+            ["release/1.x"],
+            "main",
+            {"latest": "ccf-1.0.1", "latest_same_lts": None, "next": None},
+        ),
+        # New commit on release/1.x
+        make_test_vector(
+            ["ccf-1.0.0", "ccf-1.0.1"],
+            ["release/1.x"],
+            "release/1.x",
+            {"latest": None, "latest_same_lts": "ccf-1.0.1", "next": None},
+        ),
+        # 2.0.0 release is now out
+        make_test_vector(
+            ["ccf-1.0.0", "ccf-1.0.1", "ccf-2.0.0"],
+            ["release/1.x", "release/2.x"],
+            "release/2.x",
+            {"latest": "ccf-1.0.1", "latest_same_lts": "ccf-2.0.0", "next": None},
+        ),
+        # Patch to 1.x
+        make_test_vector(
+            ["ccf-1.0.0", "ccf-1.0.1", "ccf-2.0.0", "ccf-1.0.2"],
+            ["release/1.x", "release/2.x"],
+            "ccf-1.0.3",
+            {"latest": None, "latest_same_lts": "ccf-1.0.2", "next": "ccf-2.0.0"},
```

Comment:
It looks like this is the only test case that sets `next`, so there are none which set both `latest` and `next`. Can we add one to confirm that that kind of sandwich release will work correctly (ie - testing `2.0.x` once `3.0.0` is released, and still recognising `1.0.y` as latest)?

Reply:
Good point. This is now checked too.